packages/rum-core/src/domain/rumEventsCollection/view/trackInteractionToNextPaint.spec.ts

```diff
@@ -0,0 +1,152 @@
+  resetExperimentalFeatures,
+} from '@datadog/browser-core'
+import type { TestSetupBuilder } from '../../../../test'
+import { setup } from '../../../../test'
+import type { BrowserWindow, RumPerformanceEventTiming } from '../../../browser/performanceCollection'
+import { ViewLoadingType } from '../../../rawRumEvent.types'
+import type { LifeCycle } from '../../lifeCycle'
+import { LifeCycleEventType } from '../../lifeCycle'
+import {
+  trackInteractionToNextPaint,
+  trackViewInteractionCount,
+  isInteractionToNextPaintSupported,
+} from './trackInteractionToNextPaint'
+
+describe('trackInteractionToNextPaint', () => {
+  let setupBuilder: TestSetupBuilder
+  let interactionCountStub: ReturnType<typeof subInteractionCount>
+  let getInteractionToNextPaint: () => Duration | undefined
+
+  function newInteraction(
+    lifeCycle: LifeCycle,
+    { interactionId, duration = 40 as Duration, entryType = 'event' }: Partial<RumPerformanceEventTiming>
+  ) {
+    if (interactionId) {
+      interactionCountStub.incrementInteractionCount()
+    }
+    lifeCycle.notify(LifeCycleEventType.PERFORMANCE_ENTRIES_COLLECTED, [
+      {
+        entryType,
+        startTime: relativeNow(),
+        duration,
+        interactionId,
+      },
+    ])
+  }
+
+  beforeEach(() => {
+    if (!isInteractionToNextPaintSupported()) {
+      pending('No INP support')
+    }
+
+    interactionCountStub = subInteractionCount()
+    setupBuilder = setup().beforeBuild(({ lifeCycle }) => {
+      ;({ getInteractionToNextPaint } = trackInteractionToNextPaint(ViewLoadingType.INITIAL_LOAD, lifeCycle))
+    })
+  })
+
+  afterEach(() => {
+    interactionCountStub.clear()
+  })
+
+  describe('if feature flag enabled', () => {
+    beforeEach(() => {
+      addExperimentalFeatures([ExperimentalFeature.INTERACTION_TO_NEXT_PAINT])
+    })
+
+    afterEach(() => {
+      resetExperimentalFeatures()
+    })
+
+    it('should return undefined when there are no interactions', () => {
+      setupBuilder.build()
+      expect(getInteractionToNextPaint()).toEqual(undefined)
+    })
+
+    it('should ignore entries without interactionId', () => {
+      const { lifeCycle } = setupBuilder.build()
+      newInteraction(lifeCycle, {
+        interactionId: undefined,
+      })
+      expect(getInteractionToNextPaint()).toEqual(undefined)
+    })
+
+    it('should return the p98 worst interaction', () => {
+      const { lifeCycle } = setupBuilder.build()
+      for (let index = 1; index <= 100; index++) {
+        newInteraction(lifeCycle, {
+          duration: index as Duration,
+          interactionId: index,
+        })
+      }
+      expect(getInteractionToNextPaint()).toEqual(98 as Duration)
+    })
+
+    it('should return 0 if no interaction is tracked (because the duration is below 40ms)', () => {
+      setupBuilder.build()
+      interactionCountStub.setInteractionCount(1 as Duration) // assumes an interaction happened but no PERFORMANCE_ENTRIES_COLLECTED have been triggered
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** improve test description to avoid the comment‏
```suggestion
    it('should return 0 when an interaction happened without generating a performance event (interaction duration below 40ms)', () => {
      setupBuilder.build()
      interactionCountStub.setInteractionCount(1 as Duration)
```